### DISCUSSION TITLE
Download beatmaps in lazer without Background image, Hitsounds, Storyboard and/or Skin elements

### DISCUSSION THREAD

**Discussion by Nekomosh004** [ppy/osu#18404] [Ideas] [answered]
Aside from videos, it would be awesome to have an option in lazer to download beatmaps from osu!direct without:
- Skin elements
- Storyboard elements
- Backgrounds
- Hitsounds

Unlike stable, osu!lazer no longer give us the the option to delete these manually in the game directory. I'm also one of the players who uses an external app cleaner (this is the one I'm using: https://github.com/TcNobo/TcNo-osu-Cleaner) to remove these elements from stable.

The following are the reasons why I _personally_ delete these:
- Mainly, to free up disk space;
- I prefer these deleted because sometimes these elements affect how the game perform;
- I'm using a skin and always have the options to ignore skin and hitsound elements enabled, thus, these extra beatmap elements are something I don't see/hear;
- I'm not a _fan_ of certain beatmap backgrounds and prefer them deleted; and
- Like videos, I rarely watch/play beatmaps with storyboard on.

I hope this could be implemented to osu!lazer. Thank you!

**Comment by smoogipoo** [comment]
So basically you just want to download the beatmap and audio?

Personal opinion: this feels like a slap in the face to everyone that worked on their skin/storyboards/hitsounds in their beatmap, and again personally I would go the opposite way and require playing beatmaps as the mapper intended them to be played _at least once_.

In any case, we're likely not going to be adding 16 download options to disable each one individually at your behest any time soon. This is just hard to do from a technological standpoint. But we have discussed adding on-the-fly download to osu! for cases where users don't want 20GB of disk space wasted, and for ease of transferring between PCs.

**Reply by Nekomosh004** [reply]
@smoogipoo thank you for replying c:

> Personal opinion: this feels like a slap in the face to everyone that worked on their skin/storyboards/hitsounds in their beatmap, and again personally I would go the opposite way and require playing beatmaps as the mapper intended them to be played _at least once_.

This is something I recognized after writing. But, lets be real, most players don't play with those elements turned on "_at least once_" after download. Just by looking at the top players of this game that stream live in Twitch, we can see after they download something from osu!direct, they don't do: "_Oh! Let's try playing this first with skin/storyboards/hitsounds enabled!_". Rather, they play it the optimal way. 

> In any case, we're likely not going to be adding 16 download options to disable each one individually at your behest any time soon. This is just hard to do from a technological standpoint.

I totally agree to this. Having so many download options is super bad. 
How about, after download, there's an option at the Maintenance section to "Delete XXXXX elements!"

![image](https://user-images.githubusercontent.com/82471957/170845394-86657109-4d12-4164-8a13-651aa9138d28.png)

I currently have 60,000+ beatmaps in stable, and if I haven't deleted those extra elements it would actually take a ton disk space (estimate was 4-5 GB).

This video: https://www.youtube.com/watch?v=YcYYcjFkCWM explained the disk space problems I was experiencing and solved it with the external cleaner tool.

I still hope this is something the team is reconsidering, and thank you once again for taking the time to reply c:

**Comment by peppy** [comment, marked as answer]
Not happening. You are welcome to fork the project and implement this for your personal use if you want to.
